main.go

```diff
@@ -80,6 +80,10 @@ func search(config BrokerConfig, query string) ([]shared.User, error) {
 	var results []shared.User
 
 	bodyText, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return []shared.User{}, err
```

Comment:
Results is empty at this point, so could just return it, also could theoretically return nil too.

```suggestion
		return results, err
```

Reply:
Consistency usually wins.